### ISSUE TITLE
[Apple Pay] presentApplePay() always failing

### ISSUE THREAD

**Issue by bwmx** [stripe/stripe-react-native#682]
**Describe the bug**
I have attempted to integrate Apple Pay functionality with my App using this library.

I have followed the instructions at [https://stripe.com/docs/apple-pay](url)

I have registered for apple merchant ID.

I have uploaded the cert to [https://dashboard.stripe.com/settings/payments/apple_pay](url)

When I attempt to test this (by pressing the Apple Pay button), I get the error message as seen in screenshot below. The Apple Pay dialogue is never shown to the user, presentApplePay() instantly fails.

"Cancelled" "The Payment has been cancelled". There is no debug errors, no crash or any useful logs.

I cannot find documentation that explains how to integrate Apple Pay correctly, common nuances or anything like this. Official stripe docs only account for one specific scenario for payments.

Thanks in advance anyone who can provide some help and insight.

**To Reproduce**

Ensure you are using stripe test keys (the environment where I have tested).

1. Create a new expo project.
2. Follow instructions at [https://github.com/stripe/stripe-react-native#expo](url) to integrate correctly with project.
3. wrap `<App>` with `<StripeProvider publishableKey="..." merchantIdentifier="merchant.my-app">...</StripeProvider>`
4. Integrate `<ApplePayWrapper>` and pass it the relevant parameters.
5. Build simulator build with EAS build (local or cloud)
6. Run build within simulator
7. Click the Apple Pay button. Noticing it fail with code: "Cancelled" message: "The payment has been cancelled"

```
import React, { useState } from "react";
import { View, StyleSheet, Alert, Dimensions } from "react-native";
import {
  ApplePayButton,
  useApplePay,
  ApplePay,
} from "@stripe/stripe-react-native";

type Props = {
  clientSecret: string; // paymentIntent.clientSecret - generate your own from backend
  onPaymentComplete: () => void;
  total: string; // eg "15.00"
};

const ApplePayWrapper = ({ clientSecret, onPaymentComplete, total }: Props) => {
  const [cart, setCart] = useState<ApplePay.CartSummaryItem[]>([
    { label: "Total", amount: total, type: "final" }, // Last item in array needs to reflect the total.
  ]);

  const { presentApplePay, confirmApplePayPayment, isApplePaySupported } =
    useApplePay();

  const pay = async () => {
    const { error, paymentMethod } = await presentApplePay({
      cartItems: cart,
      country: "GB",
      currency: "GBP",
      requiredBillingContactFields: ["phoneNumber", "name"],
    });

    if (error) {
      Alert.alert("presentApplePay" + error.code, error.message);
    } else {
      console.log(JSON.stringify(paymentMethod, null, 2));

      const { error: confirmApplePayError } = await confirmApplePayPayment(
        clientSecret
      );

      if (confirmApplePayError) {
        Alert.alert(
          "confirmApplePayPayment" + confirmApplePayError.code,
          confirmApplePayError.message
        );
      } else {
        // notify caller of success
        onPaymentComplete();
      }
    }
  };

  return (
    <View
      style={{
        justifyContent: "center",
        alignItems: "center",
        padding: 8,
      }}
    >
      {isApplePaySupported && (
        <ApplePayButton
          onPress={pay}
          type="plain"
          buttonStyle="black"
          borderRadius={4}
          style={styles.applePayButton}
        />
      )}
    </View>
  );
};
```
**Expected behavior**

I expect the Apple Pay dialogue/UI to pop up and allow the user to complete the payment using apple pay.

**Screenshots**
![image](https://user-images.githubusercontent.com/83704799/139109331-fd8d0eb3-6d1d-41ee-8186-27f41d108b0b.png)

**Build environment**

I am using the below version of this library:

`
"@stripe/stripe-react-native": "0.2.2",
`

**Smartphone (please complete the following information):**
 - Device: iPhone12 Simulator
 - OS:  iOS 13.X, 15.X
 - Browser: NA
 - Version: NA

**Additional context**
I have integrated Google Pay and the new CardField form correctly, all works as expected. Only trying to use ApplePay on iOS does this issue occur. So I know my clientSecret and other parameters are correct.

**Comment by charliecruzan-stripe** [comment]
This is being tested with test keys, not in your production app? Have you followed [this guide to setup a sandbox tester](https://developer.apple.com/apple-pay/sandbox-testing/)?

**Comment by charliecruzan-stripe** [comment]
Are you still seeing this issue @bwmx ?

**Comment by Abe101** [comment]
Can this issue be reopened? I am facing the same issue.
I was working on expo run and had to eject cuz I thought it may be because of expo but react native cli has the same issue

**Comment by Abe101** [comment]
> @Abe101 can you try using the the Platform Pay API available in 0.22.0? The old APIs (like `presentApplePay`) are now deprecated. Happy to take a look at this if you can provide steps to reproduce it

The platformPayPayment api did help me with the illegal invocation error when trying to pay a second time if first payment fails. It did not help me with succeeding in a payment. I may create a new issue to provide steps to reproduce the issue.
